### PR TITLE
@lennamau :added knexfile, migrated, created auth db and ran yarn

### PR DESCRIPTION
**Pull request by arunabenji29**

### CODE REVIEW

**Comment by arunabenji29** [comment]
@lennamau: get endpoint and restricted middleware

**Comment by arunabenji29** [comment]
@lennamau : added the endpoints for the user to register and login

**Comment by arunabenji29** [comment]
@lennamau : created working react app

**Comment by lennamau** [comment]
# Web Auth I
## MVP
- [x] Project Setup
- [x] Post register
- [x] Post login
- [x] Get users 
- [x] Sessions and cookie support

## Stretch
- [x] Restricted middleware and testing endpoints
- [x] React app

## What Looks Great
- Great work completing MVP and Stretch!
- Clean, uniform code
- Good file structure
- Good job with frequent commits and descriptive messages


## What Could Be Improved
- Comments in code to describe what it is doing

## Rating (1-3)
### 3

File: auth/auth-router.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {res.status(201).json(save);}
+        else{
+            res.status(404).json({message:'user could not be registered'})
+        }
+    })
+    .catch(error => {
+        res.status(500).json(error);
+    })
+});
+
```
Comment (lennamau) [inline auth/auth-router.js:25]: ✔

File: auth/auth-router.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        res.status(500).json(error);
+      });
+  });
+
+  router.delete('/logout', (req,res)=>{
+    if(req.session){
+      req.session.destroy();
+    }
+    res.status(200).json({message : 'you are successfully logged out'});
+  })
```
Comment (lennamau) [inline auth/auth-router.js:49]: ✔

File: auth/restricted.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//     }
+// }
+
+module.exports = (req,res,next)=> {
+    if(req.session && req.session.username){
+        next();
+    }
+    else {
+        res.status(401).json({message: 'cant login'});
+    }
```
Comment (lennamau) [inline auth/restricted.js:34]: ✔

File: index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    },
+    store: new KnexSessionStore({
+        knex:require('./database/dbConfig.js'),
+        tablename:'sessions',
+        sidfieldname:'sid',
+        createtable:true,
+        clearInterval: 1000 * 60 * 30,
+    }),
+};
+
```
Comment (lennamau) [inline index.js:29]: ✔

File: users/userRouter.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+const restricted = require('../auth/restricted.js')
+
+router.get('/',restricted,(req,res) => {
+    Users.find()
+    .then(users => {
+        res.json(users)
+    })
+    .catch(err => res.send(err));
+})
+
```
Comment (lennamau) [inline users/userRouter.js:13]: ✔

**Pull request merged**